tracer.go

```diff
@@ -52,38 +76,54 @@ func (t *Tracer) StartSpan(
 		option(t, s)
 	}
 
+	if !s.explicitContext && (model.SpanContext{}) != s.SpanContext {
```

Comment:
I guess the second expression is to check whether a SpanContext is empty. What is the use case? Could we make SpanContext to be created from tracing or propagation so we don't have this use case? Having an interface and making the struct implementation prove could solve this issue.

Reply:
Use case is simple... did we get a parent span through the SpanOptions... if so we either need to join the span or create a regular child span. Otherwise it is a new root span.